includes/create-theme/theme-locale.php

```diff
@@ -28,6 +28,72 @@ private static function escape_text_content( $string ) {
 
 		$string = addcslashes( $string, "'" );
 
+		// Process the string to avoid escaping inner HTML markup.
+		$p = new WP_HTML_Tag_Processor( $string );
+
+		$text   = '';
+		$tokens = array();
+		while ( $p->next_token() ) {
+			$token_type    = $p->get_token_type();
+			$token_name    = strtolower( $p->get_token_name() );
+			$is_tag_closer = $p->is_tag_closer();
+
+			if ( '#tag' === $token_type ) {
+				// Add a placeholder for the token.
+				$text .= '%s';
+				if ( $is_tag_closer ) {
+					$tokens[] = "</{$token_name}>";
+				} else {
+					// Depending on the HTML tag, we may need to process attributes so they are correctly added to the placeholder.
+					switch ( $token_name ) {
+						// Handle links.
+						case 'a':
+							$href     = esc_url( $p->get_attribute( 'href' ) );
+							$target   = empty( esc_attr( $p->get_attribute( 'target' ) ) ) ? '' : ' target="_blank"';
+							$rel      = empty( esc_attr( $p->get_attribute( 'rel' ) ) ) ? '' : ' rel="nofollow noopener noreferrer"';
+							$tokens[] = "<a href=\"{$href}\"{$target}{$rel}>";
+							break;
+						// Handle inline images.
+						case 'img':
+							$src   = esc_url( $p->get_attribute( 'src' ) );
+							$style = esc_attr( $p->get_attribute( 'style' ) );
+							$alt   = esc_attr( $p->get_attribute( 'alt' ) );
+
+							CBT_Theme_Media::add_media_to_local( array( $src ) );
+							$relative_src = CBT_Theme_Media::get_media_folder_path_from_url( $src ) . basename( $src );
+							$tokens[]     = "<img style=\"{$style}\" src=\"' . esc_url( get_stylesheet_directory_uri() ) . '{$relative_src}\" alt=\"{$alt}\">";
+							break;
+						// Handle highlights.
+						case 'mark':
+							$style    = esc_attr( $p->get_attribute( 'style' ) );
+							$class    = esc_attr( $p->get_attribute( 'class' ) );
+							$tokens[] = "<mark style=\"{$style}\" class=\"{$class}\">";
+							break;
```

Comment:
I think we should be printing out `esc_` functions here, rather than executing them. Typically the advice is to escape as late as possible, which means we should see those functions in the PHP output.

Reply:
Done here: https://github.com/WordPress/create-block-theme/pull/719/commits/678a8ef22ddecc2e0b4190dd141cfa0c4436cd1e